lib/kitchen/provisioner/habitat.rb

```diff
@@ -113,104 +108,154 @@ def create_sandbox
-        #{clean_up_screen_sessions}
-        #{clean_up_previous_supervisor}
-        echo "Running #{package_ident}."
-        #{run_package_in_background}
-        RUN
-
-        wrap_shell_code run
+        # TODO: This isn't waiting for the service to come up... should we wait? If so, how long?
+
+        # This little bit figures out what package should be loaded
+        if config[:install_latest_artifact] || !config[:artifact_name].nil?
+          # TODO: throw error and bail if there's no artifacts in the results directory
+          target_pkg = get_artifact_name
+          target_ident = "#{config[:package_origin]}/#{config[:package_name]}"
+          # TODO: This is a workaround for windows. The hart file sometimes gets copied to the
+          # %TEMP%\kitchen instead of %TEMP%\kitchen\results.
+          if windows_os?
+            target_pkg = target_pkg.gsub("results/", "") unless File.exist?(target_pkg)
+          end
+        else
+          target_pkg = package_ident
+          target_ident = package_ident
+        end
+
+        if windows_os?
+          wrap_shell_code <<~PWSH
+            if (!($env:Path | Select-String "Habitat")) {
+              $env:Path += ";C:\\ProgramData\\Habitat"
+            }
+            hab pkg install #{target_pkg} --channel #{config[:channel]} --force
+            if (Test-Path -Path "$(hab pkg path #{target_ident})\\hooks\\run") {
+              hab svc load #{target_ident} #{service_options} --force
+              $timer = 0
+              Do {
+                if ($timer -gt #{config[:service_load_timeout]}){exit 1}
+                Start-Sleep -Seconds 1
+                $timer++
+              } until( hab svc status | out-string -stream | select-string #{target_ident})
+            }
+          PWSH
+        else
+          wrap_shell_code <<~BASH
+            until sudo -E hab svc status > /dev/null
+              do
+                echo "Waiting 5 seconds for supervisor to finish loading"
+                sleep 5
+              done
+            sudo hab pkg install #{target_pkg} --channel #{config[:channel]} --force
+            if [ -f $(sudo hab pkg path #{target_ident})/hooks/run ]
+              then
+                sudo -E hab svc load #{target_ident} #{service_options} --force
+                timer=0
+                until sudo -E hab svc status | grep #{target_ident}
+                  do
+                    if [$timer -gt #{config[:service_load_timeout]}]; then exit 1; fi
+                    sleep 1
+                    $timer++
+                  done
+            fi
+          BASH
+        end
       end
 
       private
 
-      def clean_up_screen_sessions
-        return unless config[:use_screen]
+      def windows_install_cmd
+        <<~PWSH
+          if ((Get-Command hab -ErrorAction Ignore).Path) {
+            Write-Output "Habitat CLI already installed."
+          } else {
+            Set-ExecutionPolicy Bypass -Scope Process -Force
```

Comment:
Will this require Administrator privilege? I'm thinking about what we need to tell users are the prerequisites to using this on Windows

Reply:
It shouldn't since it's scoped to the process and it's running under the users context.